precompiles/assets/types.go

```diff
@@ -32,27 +35,132 @@ func (p Precompile) GetDepositWithdrawParamsFromInputs(ctx sdk.Context, args []i
-		return nil, xerrors.Errorf(exocmn.ErrInputClientChainAddrLength, len(assetAddr), types.GeneralClientChainAddrLength)
+	if uint32(len(assetAddr)) < clientChainAddrLength {
+		return nil, xerrors.Errorf(exocmn.ErrInvalidAddrLength, len(assetAddr), clientChainAddrLength)
 	}
 	depositWithdrawParams.AssetsAddress = assetAddr[:clientChainAddrLength]
 
 	stakerAddr, ok := args[2].([]byte)
 	if !ok || stakerAddr == nil {
-		return nil, xerrors.Errorf(exocmn.ErrContractInputParaOrType, 2, "[]byte", stakerAddr)
+		return nil, xerrors.Errorf(exocmn.ErrContractInputParaOrType, 2, "[]byte", args[2])
 	}
-	if len(stakerAddr) != types.GeneralClientChainAddrLength {
-		return nil, xerrors.Errorf(exocmn.ErrInputClientChainAddrLength, len(assetAddr), types.GeneralClientChainAddrLength)
+	if uint32(len(stakerAddr)) < clientChainAddrLength {
+		return nil, xerrors.Errorf(exocmn.ErrInvalidAddrLength, len(stakerAddr), clientChainAddrLength)
 	}
 	depositWithdrawParams.StakerAddress = stakerAddr[:clientChainAddrLength]
 
 	opAmount, ok := args[3].(*big.Int)
 	if !ok || opAmount == nil || opAmount.Cmp(big.NewInt(0)) == 0 {
-		return nil, xerrors.Errorf(exocmn.ErrContractInputParaOrType, 3, "*big.Int", opAmount)
+		return nil, xerrors.Errorf(exocmn.ErrContractInputParaOrType, 3, "*big.Int", args[3])
 	}
 	depositWithdrawParams.OpAmount = sdkmath.NewIntFromBigInt(opAmount)
 
 	return depositWithdrawParams, nil
 }
+
+func (p Precompile) ClientChainInfoFromInputs(_ sdk.Context, args []interface{}) (*types.ClientChainInfo, error) {
+	if len(args) != 5 {
+		return nil, xerrors.Errorf(cmn.ErrInvalidNumberOfArgs, 5, len(args))
+	}
+	clientChain := types.ClientChainInfo{}
+	clientChainID, ok := args[0].(uint32)
+	if !ok {
+		return nil, fmt.Errorf(exocmn.ErrContractInputParaOrType, 0, "uint32", args[0])
+	}
+	clientChain.LayerZeroChainID = uint64(clientChainID)
+
+	addressLength, ok := args[1].(uint8)
+	if !ok || addressLength == 0 {
+		return nil, fmt.Errorf(exocmn.ErrContractInputParaOrType, 1, "uint8", args[1])
+	}
+	if addressLength < types.MinClientChainAddrLength {
+		return nil, xerrors.Errorf(exocmn.ErrInvalidAddrLength, addressLength, types.MinClientChainAddrLength)
+	}
+	clientChain.AddressLength = uint32(addressLength)
+
+	name, ok := args[2].(string)
+	if !ok {
+		return nil, xerrors.Errorf(exocmn.ErrContractInputParaOrType, 2, "string", args[2])
+	}
+	if name == "" || len(name) > types.MaxChainTokenNameLength {
+		return nil, xerrors.Errorf(exocmn.ErrInvalidNameLength, name, len(name), types.MaxChainTokenNameLength)
+	}
+	clientChain.Name = name
+
+	metaInfo, ok := args[3].(string)
+	if !ok {
+		return nil, xerrors.Errorf(exocmn.ErrContractInputParaOrType, 3, "string", args[2])
+	}
+	if metaInfo == "" || len(metaInfo) > types.MaxChainTokenMetaInfoLength {
+		return nil, xerrors.Errorf(exocmn.ErrInvalidMetaInfoLength, metaInfo, len(metaInfo), types.MaxChainTokenMetaInfoLength)
+	}
+	clientChain.MetaInfo = metaInfo
+
+	signatureType, ok := args[4].(string)
```

Comment:
should we use an enum value instead of string if know common signature types existed. Of course we could also leave it for future PRs if we haven't researched into this

Reply:
Using the string type might be clearer when querying or setting values. However, using enum can reduce gas costs. This can be decided later when this field is actually used.